### ISSUE TITLE
Symfony 3 compatible version

### ISSUE THREAD

**Issue by caciobanu** [doctrine/DoctrineMongoDBBundle#360]
Is it possible to release a version of the bundle which would be compatible with symfony 3? Are there anymore issues that need to be tackled before a release is possible ?

As far as I can tell from the code in master the compatibility changes have been made but a new version has not yet been released.

**Comment by malarzm** [comment]
There is one thing left to fix (namely #357 which #358 tried to address) - other than that we're good to go, I just haven't got enough time to check why #358 failed. If you could give us a hand on this that'd be much appreciated!

**Comment by caciobanu** [comment]
I think that the issue comes from here: https://github.com/doctrine/DoctrineMongoDBBundle/blob/master/DependencyInjection/Compiler/DoctrineMongoDBMappingsPass.php

I think that driver definition should look like this:
$driver = new Definition('Doctrine\ODM\MongoDB\Mapping\Driver\YamlDriver', array($locator, , '.mongodb.yml'));

**Comment by malarzm** [comment]
Hmmm this place does look interesting but would also mean that simplified drivers that were in bundle were not used at all?

**Comment by caciobanu** [comment]
Unfortunately I did not made any tests just looked through the code (limited internet access for now).

**Comment by malarzm** [comment]
No problem, just spoke out what came to my mind :)

**Comment by caciobanu** [comment]
The issue seems much bigger. I am at the moment looking at Symfony\Bridge\Doctrine\DependencyInjection\AbstractDoctrineExtension::registerMappingDrivers(). I think that to keep the old prefix we have to undo the deprecation of the mapping drivers from the bundle and use those in the config.xml instead of the common doctrine simplified versions.

**Comment by caciobanu** [comment]
What I've discovered is kinda hard to explain but here it goes:

At some point DoctrineMongoDBBundle registers for each document manager the mapping drivers. This can be seen in https://github.com/doctrine/DoctrineMongoDBBundle/blob/master/DependencyInjection/DoctrineMongoDBExtension.php#L328, more specific line 335.

The method registerMappingDrivers() exists in the extended class from the symfony doctrine bridge: https://github.com/symfony/symfony/blob/master/src/Symfony/Bridge/Doctrine/DependencyInjection/AbstractDoctrineExtension.php#L190 

The interesting lines in AbstractDoctrineExtension are 201-202: the name of the service for each metadata driver (xml, yml, annotation, php) is built from the name of the document manager. Because there is no registered service for that specific document manager it executes lines 216-220 where you can see that the service definition is built from scratch but an essential parameter is missing: the file extension. That is why when using the simplified mapping drivers from the ODM and not from the Bundle the extension is **.mongo-odm.xxx**.

To fix the issue we have to change https://github.com/doctrine/DoctrineMongoDBBundle/blob/master/Resources/config/mongodb.xml#L37 https://github.com/doctrine/DoctrineMongoDBBundle/blob/master/Resources/config/mongodb.xml#L38 to:
```
<parameter key="doctrine_mongodb.odm.metadata.xml.class">Doctrine\Bundle\MongoDBBundle\Mapping\Driver\XmlDriver</parameter>
<parameter key="doctrine_mongodb.odm.metadata.yml.class">Doctrine\Bundle\MongoDBBundle\Mapping\Driver\YamlDriver</parameter>
```

**Comment by caciobanu** [comment]
I've added a pull request also, addressing this issue: https://github.com/doctrine/DoctrineMongoDBBundle/pull/361

**Comment by malarzm** [comment]
Thanks a bunch @caciobanu for investigation!

**Comment by caciobanu** [comment]
It was a pleasure doing it.

**Comment by malarzm** [comment]
[Released 3.1.0](https://github.com/doctrine/DoctrineMongoDBBundle/releases/tag/3.1.0) :)

**Comment by caciobanu** [comment]
Thank you @malarzm! Awesome news.
